components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -183,6 +196,8 @@ private void handleClientError(ActionInvocationResponse.Builder builder, HttpEnt
                     " is not of the expected error response format. ", e);
             builder.errorLog("Unexpected error response received for the status code: " + statusCode + ". "
                     + e.getMessage());
+            ActionInvocationErrorResponse invalidResponse = handleInvalidResponse();
+            builder.response(invalidResponse);
```

Comment:
Seems like this part is common for all handling scenarios. Shall just throw the error from here and handle this error from the handleResponse method